### PR TITLE
Add Results of Scaling Experiments

### PR DESCRIPTION
**Pull request by mali-git**
# What does this PR do?

This PR add the results of our base scaling experiments.

## General Changes
* 

## Breaking Changes
* None.

## Checklist before submitting final PR
- [x] My PR is minimal and addresses one issue in isolation
- [x] I have merged the latest version of the target branch into this feature branch
- [x] I have reviewed my own code w.r.t. correct implementation, missing type hints, proper documentation, etc.
- [x] I have run a sample config for model training
- [x] I have checked that all tests run through (`python tests/tests.py`)
- [x] I have updated the internal changelog (`CHANGELOG_DEV.md`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -163,6 +163,76 @@ in our `pyproject.toml`, we can start only main with `modalities` (which does no
 
 Alternatively, directly use `src/modalities/__main__.py do_stuff --config_file_path config_files/config.yaml --my_cli_argument 3537`.
 
+# Scaling Experiments
```
Comment (flxst) [inline README.md:166]:
```suggestion
## Scaling Experiments
```
top header level in the readme is 2

File: README.md [hunk 2]
```diff
@@ -163,6 +163,76 @@ in our `pyproject.toml`, we can start only main with `modalities` (which does no
...
+
+In the following, you can find the results of our scaling experiments performed on two HPC centers, namely [Leonardo Booster](https://leonardo-supercomputer.cineca.eu/hpc-system/) and [MareNostrum 5](https://www.bsc.es/ca/marenostrum/marenostrum-5). 
+
+In a first step, we explored a **limited** set of different configurations (batch size, gradient accumulation steps, etc.) to get our baseline results. In a second step, we will focus on optimizing these configurations to maximize throughput.
+
+
+## Leonardo Booster  - NVIDIA A100 64GB
```
Comment (flxst) [inline README.md:173]:
```suggestion
### Leonardo Booster  - NVIDIA A100 64GB
```
see comment above

File: README.md [hunk 3]
```diff
@@ -163,6 +163,76 @@ in our `pyproject.toml`, we can start only main with `modalities` (which does no
...
+| 6.7 | 8 | 9.28 | 1 | 2 | 16 | 4096 | BF_16 | FULL_SHARD | False |  A100  | 0.6867 |
+| 6.7 | 16 | 18.35 | 1 | 2 | 32 | 4096 | BF_16 | FULL_SHARD | False |  A100  | 0.6789 |
+| 6.7 | 32 | 36.65 | 1 | 2 | 64 | 4096 | BF_16 | FULL_SHARD | False |  A100  | 0.6782 |
+| 6.7 | 64 | 72.72 | 1 | 2 | 128 | 4096 | BF_16 | FULL_SHARD | False |  A100  | 0.6727 |
+| 6.7 | 128 | 131.59 | 1 | 2 | 256 | 4096 | BF_16 | FULL_SHARD | False |  A100  | 0.6086 |
+| 6.7 | 256 | 225.24 | 1 | 2 | 512 | 4096 | BF_16 | FULL_SHARD | False |  A100  | 0.5209 |
+
+Further scaling results can be found at [Leonardo Booster Scaling Experiments](https://github.com/Modalities/modalities/blob/scaling_experiments/data/scaling_experiments/scaling_leonardo.md)
+
+## MareNostrum 5 - NVIDIA H100 64GB
```
Comment (flxst) [inline README.md:203]:
```suggestion
### MareNostrum 5 - NVIDIA H100 64GB
```
see comment above

**Review by flxst — APPROVED** [review]
LGTM! Left minor suggestions for improvement. And I wonder if we should really have the tables and plot in the folder `data`, where they get mixed up with input data. Perhaps move them to `docs` instead (assuming that we will also include them in the docs later)?

**Comment by mali-git** [comment]
> LGTM! Left minor suggestions for improvement. And I wonder if we should really have the tables and plot in the folder `data`, where they get mixed up with input data. Perhaps move them to `docs` instead (assuming that we will also include them in the docs later)?

Moved data into docs, please see bb03bec6bfc4ab7b507f24e413223c70e5e6cb7d.

**Pull request merged**